server/src/db.rs

```diff
@@ -45,23 +46,30 @@ pub struct NewGameEvent {
     pub event: serde_json::Value,
 }
 
-impl<DB: Backend> FromSql<Integer, DB> for GameId
+impl<DB: Backend> FromSql<Text, DB> for GameId
 where
-    i32: FromSql<Integer, DB>,
+    String: FromSql<Text, DB>,
+    //*const str: FromSql<Text, DB>,
 {
     fn from_sql(bytes: DB::RawValue<'_>) -> diesel::deserialize::Result<Self> {
-        <i32 as FromSql<Integer, DB>>::from_sql(bytes).map(|n| GameId(n as u32))
+        // having to allocate a string here brings me great sadness
```

Comment:
If you think it's worth it we can remove the prevent unsafe thing.

Reply:
no, not worth it unless performance suffers